app/decorators/controllers/solidus_bolt/spree_checkout_controller/add_addresses_to_bolt.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module SpreeCheckoutController
+    module AddAddressesToBolt
+      def finalize_order
+        spree_current_user&.addresses&.each do |address|
```

Comment:
I think we should call this service only when the user has a bolt account, access token is present, and bolt payment method is used.

Reply:
fixed in https://github.com/nebulab/solidus_bolt/pull/96/commits/9b96b27b52d35163300716f53ad0c61187da3d7a